testing/test_doctest.py

```diff
@@ -475,6 +475,85 @@ def foo():
                                     "--junit-xml=junit.xml")
         reprec.assertoutcome(failed=1)
 
+    def _run_doctest_report(self, testdir, format):
```

Comment:
I appreciate the well written tests!

I think we could move all those tests to a separate test class (`TestReportOptions` maybe?) to better group them and this utility function together. What do you think?